peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
+consistent with Python's raw string behavior.
 
-.. code-block:: python
+As with regular template strings, interpolations in raw template strings are 
+processed normally, allowing for the combination of raw string behavior and 
+dynamic content.
 
-    class InterpolationConcrete(NamedTuple):
-        getvalue: Callable[[], Any]
-        expr: str
-        conv: Literal['a', 'r', 's'] | None = None
-        format_spec: str | None = None
 
 Interpolation Expression Evaluation
 -----------------------------------
 
-Expression evaluation for interpolations is the same as in :pep:`498#expression-evaluation`,
-except that all expressions are always implicitly wrapped with a ``lambda``:
+Expression evaluation for interpolations is the same as in :pep:`498#expression-evaluation`:
 
     The expressions that are extracted from the string are evaluated in the context
-    where the tag string appeared. This means the expression has full access to its
+    where the template string appeared. This means the expression has full access to its
     lexical scope, including local and global variables. Any valid Python expression
     can be used, including function and method calls.
 
-However, there's one additional nuance to consider, `function scope
-<https://docs.python.org/3/reference/executionmodel.html#resolution-of-names>`_
-versus `annotation scope
-<https://docs.python.org/3/reference/executionmodel.html#annotation-scopes>`_.
-Consider this somewhat contrived example to configure captions:
-
-.. code-block:: python
-
-    class CaptionConfig:
-        tag = 'b'
-        figure = f'<{tag}>Figure</{tag}>'
+Template strings are evaluated eagerly from left to right, just like f-strings. This means that
+interpolations are evaluated immediately when the template string is processed, not deferred
+or wrapped in lambdas.
 
-Let's now attempt to rewrite the above example to use tag strings:
 
-.. code-block:: python
+Exceptions
+----------
 
-    class CaptionConfig:
-        tag = 'b'
-        figure = html'<{tag}>Figure</{tag}>'
+Exceptions raised in t-string literals are the same as those raised in f-string 
+literals.
 
-Unfortunately, this rewrite doesn't work if using the usual lambda wrapping to
-implement interpolations, namely ``lambda: tag``. When the interpolations are
-evaluated by the tag function, it will result in ``NameError: name 'tag' is not
-defined``. The root cause of this name error is that ``lambda: tag`` uses function scope,
-and it's therefore not able to use the class definition where ``tag`` is
-defined.
 
-Desugaring how the tag string could be evaluated will result in the same
-``NameError`` even using f-strings; the lambda wrapping here also uses function
-scoping:
+Interleaving of ``Template.args``
+---------------------------------
 
-.. code-block:: python
+As a detail of the ``Template`` type, the ``args`` attribute is a sequence that
+alternates between string literals and ``Interpolation`` instances. Specifically:
```

Comment:
Is this an implementation detail or formally specified?

Reply:
Formally specified.